### PR TITLE
[JS/C API] Expose zeroFilledMemory option for JS and C API

### PR DESCRIPTION
**Pull request by MaxGraey**

### CODE REVIEW

**Comment by MaxGraey** [comment]
could somebody review this simple PR?

File: src/binaryen-c.h [hunk 1]
```diff
@@ -2284,6 +2284,13 @@ BINARYEN_API bool BinaryenGetLowMemoryUnused(void);
 // when optimizing. Applies to all modules, globally.
 BINARYEN_API void BinaryenSetLowMemoryUnused(bool on);
 
+// Gets whether that an imported memory will be zero-initialized speculation.
```
Comment (kripken) [inline src/binaryen-c.h:2287]:
```suggestion
// Gets whether to assume that an imported memory is zero-initialized.
```

File: src/binaryen-c.h [hunk 2]
```diff
@@ -2284,6 +2284,13 @@ BINARYEN_API bool BinaryenGetLowMemoryUnused(void);
...
+BINARYEN_API bool BinaryenGetZeroFilledMemory(void);
+
+// Enables or disables whether that an imported memory will be
+// zero-initialized speculation.
```
Comment (kripken) [inline src/binaryen-c.h:2291]:
```suggestion
// Enables or disables whether to assume that an imported memory is zero-initialized.
```

**Review by kripken — COMMENTED** [review]
lgtm %  comments

**Review by kripken — APPROVED** [review]

**Comment by kripken** [comment]
Looks like there is a small lint issue before landing. Otherwise looks good.

**Pull request merged**